### PR TITLE
Java: Allow ChatCompletion to be used as a TextCompletion

### LINKED ISSUE #2020: Java: chat completion: implement text completion interface

### PR DESCRIPTION
**Pull request by johnoliver**
If a TextCompletion is not available for that operation, the kernel will use a ChatCompletion service if available.

Fixes #2020

### CODE REVIEW

**Review by dsgrieve — APPROVED** [review]

File: java/semantickernel-connectors-parent/semantickernel-connectors/src/main/java/com/microsoft/semantickernel/connectors/ai/openai/chatcompletion/OpenAIChatCompletion.java [hunk 1]
```diff
@@ -26,6 +28,14 @@ public OpenAIChatCompletion(OpenAIAsyncClient client, String modelId) {
         super(client, modelId);
     }
 
+    @Override
+    public Mono<List<String>> completeAsync(
```
[the remark this answers is not in the document]
Comment (johnoliver) [reply java/semantickernel-connectors-parent/semantickernel-connectors/src/main/java/com/microsoft/semantickernel/connectors/ai/openai/chatcompletion/OpenAIChatCompletion.java:32]: non null is the assumed default if not explicitly nullable, for most of the code base it is assumed that if not annotated then its non-null

File: java/semantickernel-api/src/main/java/com/microsoft/semantickernel/textcompletion/CompletionRequestSettings.java [hunk 2]
```diff
@@ -124,6 +124,18 @@ public CompletionRequestSettings() {
         this(0, 0, 0, 0, 256, 1, "", new ArrayList<>());
     }
 
+    public CompletionRequestSettings(CompletionRequestSettings requestSettings) {
+        this(
+                requestSettings.getTemperature(),
```
[the remark this answers is not in the document]
Comment (johnoliver) [reply java/semantickernel-api/src/main/java/com/microsoft/semantickernel/textcompletion/CompletionRequestSettings.java:129]: Unfortunately as this is a constructor and it is calling `this` it is hard to defend against null without sending the request to some other static method

File: java/semantickernel-connectors-parent/semantickernel-connectors/src/main/java/com/microsoft/semantickernel/connectors/ai/openai/chatcompletion/OpenAIChatCompletion.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (dsgrieve) [reply java/semantickernel-connectors-parent/semantickernel-connectors/src/main/java/com/microsoft/semantickernel/connectors/ai/openai/chatcompletion/OpenAIChatCompletion.java:32]: True. But my point is, I see @Nullable in ChatCompletion and not in TextCompletion and I think "Oh, the implementor must have made a mistake and it should be @Nullable". This is a case where it helps document the code. 

As an aside, my IDE doesn't warn me unless it's explicitly @Nonnull.

File: java/semantickernel-api/src/main/java/com/microsoft/semantickernel/textcompletion/CompletionRequestSettings.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (markwallace-microsoft) [reply java/semantickernel-api/src/main/java/com/microsoft/semantickernel/textcompletion/CompletionRequestSettings.java:129]: We can do `(requestSettings != null) ? requestSettings.getTemperature() : DEFAULT_TEMPERATURE,`

**Review by markwallace-microsoft — APPROVED** [review]
LGTM once you have the comments resolved we can merge

File: java/semantickernel-api/src/main/java/com/microsoft/semantickernel/textcompletion/CompletionRequestSettings.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (johnoliver) [reply java/semantickernel-api/src/main/java/com/microsoft/semantickernel/textcompletion/CompletionRequestSettings.java:129]: I have added NonNull annotation

**Comment by johnoliver** [comment]
Should be good now

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**